### ISSUE TITLE
problem while using openMVG_main_ColHarmonize

### ISSUE THREAD

**Issue by miller233** [openMVG/openMVG#990]
Hi,
  Is it possible to use openMVG_main_ColHarmonize to removing the uneven illumination? I tested a data set, the result seams eccentric. the images near the ref image looks fine, but the images faraway from the ref image become odd (enven pure yellow or white). what could be the reason lead to this? is there any suggestion?

miller

**Comment by pmoulon** [comment]
If results looks weird it's certainly that some of the matches betweem the images still have some inconsistencies.

1. Can you tell me which `--selectionMethod ` method did you choose?
I recommend to use `- Matched Points: 1`. `- VLD Segment: 2` can also enhance results but sometime it is a bit slow.

Here the thing you can try in order to have less outlier corresponding regions:

- try the matches constrained by homographies (`-g h` option on `ComputeMatches`).

The method have proven to work nice on panoramic image like sequence. In multiple view settings   often it works less good.

**Comment by miller233** [comment]
Three different selectionMethod have been test, and all of them result in weird color. The image is taken by UAV. This is some of the result images using VLD Segment:
![img](https://user-images.githubusercontent.com/25821619/28397029-975ef000-6d31-11e7-92f1-4a9148c5ad4d.PNG)

**Comment by rjanvier** [comment]
Hello I'm developing a new color harmonization method for openMVG. The code is not ready to be published yet because it's still a bit raw and untested. If you want I can make a test on your dataset or a subset of your dataset.

**Comment by miller233** [comment]
Hi, @rjanvier. Thanks for your help. But it seams not allowed to share UVA images according to the law of my country. I would like to test some open dataset, if this weird result happened in open dataset I would send you the URL of the dataset. But I need some time to perform this work (maybe some days later). I will give you some feedback. And It would be great if you can let me konw when your new color harmonization method is ready to release.
Miller

**Comment by pmoulon** [comment]
@miller233 It would be interesting also to know the graph relation that you are using.
If we see such a divergence it certainly because some of the images are not strongly linked.
- if you have graphviz you should have a graph input_graph_poor_supportRemoved.svg on disk. If you can share it we will have a better idea of the graph shape (pairwise relationship that are used for the minimization).

**Comment by pmoulon** [comment]
I invite you also to take a look to "Color Correction for Image-Based Modeling in
the Large" http://www.cse.ust.hk/~szhu/publications/color_correction.pdf that propose an update of the method introduced in `openMVG_main_ColHarmonize [1]`

[1] Moulon, P., Duisit, B., Monasse, P.: Global multiple-view color consistency. In: Conference on Visual Media Production (CVMP). (2013)

**Comment by miller233** [comment]
Thanks for recommend that paper. I will read carefully.
This is the graph file for the dataset I tested. the dataset can connect each other (since the sfm results looks well) but the landmarks points is sparse. This may be the reason lead to this weird result.
[input_graph_poor_supportRemoved.txt](https://github.com/openMVG/openMVG/files/1164765/input_graph_poor_supportRemoved.txt)

**Comment by pmoulon** [comment]
As you can see two part are more connected and the rest of the connection is quite sparse.

![input_graph_poor_supportremoved txt](https://user-images.githubusercontent.com/3235050/28454735-9aabf12e-6dfc-11e7-8080-4693121975d1.png)

**Comment by pmoulon** [comment]
Closed due to inactivity.

**Comment by yujie-yang** [comment]
Sorry to bother you. I also encountered a similar issue. I used openMVG_main_ColHarmonize to a set of 30 images taken from multiple angles, but the results were not satisfactory, and it seems like the illumination correction did not work as expected. Additionally, some images had strange color shifts, with the overall tone appearing yellowish, which deviates significantly from the desired outcome. I tried all three methods in the selection process, but the results were consistent with this issue.
I also try the matches constrained by homographies (-g h option on openMVG_main_GeometricFilter), it still didn't work.
Here's my geometric_matches.svg.
![企业微信截图_16969049147220](https://github.com/openMVG/openMVG/assets/68651860/03ac9c11-bc9e-422e-af7f-bad41630a81e)
If it's the problem of some of the images are not strongly linked, what should I do?
